### PR TITLE
Few improvements found when rehearsing advanced course day 1

### PR DESCRIPTION
**Pull request by sponce**

### CODE REVIEW

File: talk/objectorientation/typecasting.tex [hunk 1]
```diff
@@ -1,66 +1,82 @@
...
   \end{block}
+\end{frame}
 
-  \vspace{-5mm}
+\begin{frame}[fragile]
+  \frametitlecpp[98]{Type casting example}
+  \small
+  \begin{exampleblock}{}
+    \begin{cppcode*}{gobble=2}
+      struct A{ virtual ~A(){} } a;
```
Comment (bernhardmgruber) [inline talk/objectorientation/typecasting.tex:28]:
```suggestion
      struct A{ virtual ~A() = default; } a;
```

File: talk/objectorientation/typecasting.tex [hunk 2]
```diff
@@ -1,66 +1,82 @@
...
+  \begin{block}{5 types of casts in \cpp\ - These 2 should be used}
     \begin{itemize}
-    \item \mintinline{c++}{static_cast<Target>(arg)}: Convert type if the static types allow it
-    \item \mintinline{c++}{dynamic_cast<Target>(arg)}: Check if object at address of ``\mintinline{c++}{arg}'' is convertible to type \mintinline{c++}{Target}.
-          \alt<1>{Throws \mintinline{c++}{std::bad_cast} if it's not.}{Return \mintinline{c++}{nullptr} if it's not.}
+    \item \mintinline{c++}{static_cast<Target>(arg)}
+      \begin{itemize}
+      \item converts type if the static types allow it
+      \item including using implicit conversion
+      \item essentially compile time
```
[the remark this answers is not in the document]
Comment (sponce) [reply talk/objectorientation/typecasting.tex:11]: Yes, that's precisely what the "essentially" is there for. I'll detail verbally. In the case of basci conversion (no function calls) and with no multiple inheritance, it's still only compile time.

File: talk/objectorientation/typecasting.tex [hunk 3]
```diff
@@ -1,66 +1,82 @@
...
+      \end{itemize}
+    \item \mintinline{c++}{dynamic_cast<Target>(arg)}
+      \begin{itemize}
+      \item checks if object at address of ``\mintinline{c++}{arg}'' is convertible to \mintinline{c++}{Target}
+      \item throws \mintinline{c++}{std::bad_cast} or returns  \mintinline{c++}{nullptr} if it's not
+      \item 100\% run time
```
[the remark this answers is not in the document]
Comment (sponce) [reply talk/objectorientation/typecasting.tex:17]: Well, you could even argue that the compiler can convert it to a static_cast when it sees that there is no need for the dynamic one. Ok, will replace with "essentially run time" if it's fine with you

File: talk/objectorientation/typecasting.tex (hunk 2 above)
[the remark this answers is not in the document]
Comment (bernhardmgruber) [reply talk/objectorientation/typecasting.tex:11]: ok, fine!

**Pull request merged**